lib/jxl/dec_external_image.cc

```diff
@@ -350,7 +350,7 @@ Status ConvertToExternal(const jxl::ImageBundle& ib, size_t bits_per_sample,
             const float* JXL_RESTRICT row_in[4];
             size_t c = 0;
             for (; c < color_channels; c++) {
-              row_in[c] = color->PlaneRow(c, y);
+              row_in[c] = color->PlaneRow(color_channels == 1 ? 1 : c, y);
```

Comment:
Why 1 exactly? Wouldn't all 3 channels be the same when the image is grayscale?

Reply:
The channels are in Cb, Y, Cr order (it's that way to match XYB order so chroma from luma can be used) and in the case of grayscale the Cb and Cr channels are not used.

Reply:
But this is in ConvertToExternal, nothing specifies what color format the image is in.

Now, if rgb and ycbcr are the only possibilities, where both happen to work (since for grayscale, rgb's g, and component 1 of Cb,Y,Cr it happens to match), then this is fine, but are we sure of this?

Reply:
Agreed this is probably not the right place to fix it.